File: 백준/Bronze/11653. 소인수분해/소인수분해.py
import sys
input = sys.stdin.readline

N = int(input())
result = []

def fin(N):
    for i in range(2, N+1):
        if N % i == 0:
            a = i
            result.append(a)
            break

    b = N // a
    if b > a:
        fin(b)
    elif b != 1:
        result.append(b)

if N != 1:
    fin(N)

    for i in result:
        print(i)

File: 백준/Bronze/11653. 소인수분해/소인수분해.js
let fs = require('fs')
let input = fs.readFileSync("/dev/stdin").toString().trim().split(" ")
// let input = fs.readFileSync("./EX.txt").toString().trim().split(" ")
const N = parseInt(input)
let arr = []
function fin(N){
    for (let i = 2; i <= N; i++) {
        if (N % i == 0) {
            a = i
            arr.push(a)
            break
        }
    }
    b = parseInt(N / a)
    if (b > a) {
        fin(b)
    } else if (b != 1) {
        arr.push(b)
    }
}

if (N != 1) {
    fin(N)
    arr.forEach(x => console.log(x))
}